### ISSUE TITLE
[evergreen] Depending on not existant tooling is problematic

### ISSUE THREAD

**Issue by frivoal** [w3c/w3process#301]
TLDR: Desiring future tooling that could make our life better is good. Designing a process that cannot work unless we have tooling that currently doesn't exist is dangerous.

In the [section on how to publish evergreen RECs](https://w3c.github.io/w3process/evergreen/#evergreen-rec)The Evergreen Process places a number of requirements on the Editor,  **every time they want to publish** and **for each feature**:
> * must provide a designation indicating the level of implementation experience
> * must provide a designation whether the feature has been in the spec for at least 180 days (giving some indication that the feature has received sufficient review).
> * must provide a designation whether the feature has appeared in an Evergreen Recommendation snapshot and is therefore covered by the Evergreen Review Draft Licensing Obligations.
> * should provide information as to whether the feature has implementation commitments

Done manually, this is a very large amount of work. Large enough to be unrealistic. When discussing this, the explanation provided was that this is expected to be alleviated by tooling, making the workload a non issue.

However, the tooling currently does not exist, and it is not clear that it can realistically be expected to come into existence within the forseeable future (more details below).

If we intent to wait until these tools exist before rolling out evergreen, we may have to wait a long time, possibly forever. If we intent to roll it out before they do, the process will be unworkably heavy, and working groups will either not use it, or use it without abiding by the requirements. Both are bad.

----

More details about why the required tooling neither exists currently, nor in some cases can be expected to come into existence soon:

1.
    > * must provide a designation indicating the level of implementation experience

    While it is unlikely that the implementation status of every feature in the spec will change at every publication, especially if they are frequent, even merely verifying that it has not changed is a non trivial amount of work.

    Not only is there no tool today that can inject an annotation carrying this information into the specification, this is not information that anyone has collected currently. The closest things we have at the moment, but which fall short of the requirement, are:
    * wpt test results. As wpt.fyi itself claims at the top of the page, in red:
        > The reported results do not necessarily reflect the true capabilities of each web browser, so they should not be used evaluate or compare feature support
       
        The problems with using wpt results as a source of information on the level of implementation include:
        * No indication on how complete the coverage of the specification is. Significant aspects of the spec may lack tests, and this cannot be reported by wpt. Not only is there no existing data on this, there is no agreement that building tooling to automatically assess this is possible.
        * No reliable "per feature" grouping of test, but the requirement on the Evergreen WG is to document "per feature. wpt tests are always tagged per specification, and some of the time per section of the specification, but no finer tagging exists.
        * If the definition of a feature for which tests already exist is modified, there is no automated way to know whether the corresponding tests have been updated or not. They may have been, but wpt may report passing tests that no longer correspond to the specification.
    * caniuse.com support information. Problems with this include:
        * not all features are covered
        * maintained on a volunteer basis by third parties out of control of the WG, and not always up to date, or even not always correct.
        * The granularity of the information available is insufficient for our purpose. There is no strict rule on how fine-grained the caniuse documentation has to be, but it is fairly typical for caniuse to treat an entire specification as being a single "feature", even for large and complex specifications (e.g. [css-grid](https://caniuse.com/#feat=css-grid), [Payment Request API](https://caniuse.com/#feat=payment-request), [mathML](https://caniuse.com/#feat=mathml))
        * If the definition of a feature for which a caniuse entry already exist is modified, there is no automated way to know whether the caniuse entry has been updated or not. It may have been, but caniuse may report support for implementations that no longer correspond to the specification.
    * MDN. Largely has the same limitations as caniuse, even though it tends to have more fine grained information

    This is not to say that info panels based on these sources aren't useful. They are. I support using tooling that injects this information into specifications (see [css-ui-4](https://drafts.csswg.org/css-ui-4/) for an example of a spec with test and caniuse info). But the information they can provide falls short of what the Evergreen Process requires, and I do not believe I have seen concrete proposals for something that would satisfy the requirements.

2. > must provide a designation whether the feature has been in the spec for at least 180 days

    Presumably, the workload for this can be significantly reduced by simply indicating in the specification the date the feature was introduced, rather than a duration, which would then need to be updated at each publication. However, this would imply a specification littered with dates.

    More over, since this information is meant to be used to assess whether the feature has had the chance to get sufficient review, this date should be updated when the definition of a previously specified feature is modified, or at least, the modification should be annotated with a date. Overtime, the amount of dates carried in the specification may become very large, threatening the readability of the document.

    It is reasonable to think that this problem can be solved thanks to good design of how the annotations look like, or by tooling that could reveal hidden-by-default date information, but no such design / tooling has been proposed yet.

    Maybe bikeshed / respec could be augmented to support some kind of date-of-last-change attribute in the markup, and generate non visually intrusive markings for things that have been in the spec for less than 180 days, and no markings for things that have been there for more. And then the Editor could, once in a while, purge old dates.

    Unlike point 1, I have no particular reason to think this problem may be exceptionally hard to solve. Even then, it hasn't been solved yet.

3. > * must provide a designation whether the feature has appeared in an Evergreen Recommendation snapshot and is therefore covered by the Evergreen Review Draft Licensing Obligations.

    Largely the same problem as the 180 days markings. Not necessarily hard conceptually, but no tooling exists for this today.

4 > * should provide information as to whether the feature has implementation commitments

    This too lacks existing tooling to assist keeping that information up to date. More over, unlike 2 and 3, this relies on information that exists outside of the specification itself, and for which there is no central and authoritative repository. (nor is there any clear definition of what constitutes "implementation commitment", and whether that differs from "implementation interest")

Out of these, 2, 3, and 4, while annoying, maybe be surmountable. However, I do not think that 1 can be performed manually at the required frequency and level of precision, nor that it can be automated at the required level of precision any time soon.

**Comment by frivoal** [comment]
agreed to close on Jan 29th call, as a separate Evergreen track is no longer being pursued.
